model/atmosphere/subgrid_scale_physics/microphysics/src/icon4py/model/atmosphere/subgrid_scale_physics/microphysics/single_moment_six_class_gscp_graupel.py

```diff
@@ -0,0 +1,2785 @@
+            math.pi * phy_const.water_density / 6.0 * _n0r * math.gamma(self.config.rain_mu + 4.0)
+        )  # pre-factor
+
+        power_law_exponent_for_rain_mean_fall_speed: ta.wpfloat = 0.5 / (self.config.rain_mu + 4.0)
+        power_law_coeff_for_rain_mean_fall_speed: ta.wpfloat = (
+            130.0
+            * math.gamma(self.config.rain_mu + 4.5)
+            / math.gamma(self.config.rain_mu + 4.0)
+            * _ar ** (-power_law_exponent_for_rain_mean_fall_speed)
+        )
+
+        precomputed_evaporation_alpha_exp_coeff: ta.wpfloat = (self.config.rain_mu + 2.0) / (
+            self.config.rain_mu + 4.0
+        )
+        precomputed_evaporation_alpha_coeff: ta.wpfloat = (
+            2.0
+            * math.pi
+            * microphy_const.diffusion_coeff_for_water_vapor
+            / microphy_const.howell_factor
+            * _n0r
+            * _ar ** (-precomputed_evaporation_alpha_exp_coeff)
+            * math.gamma(self.config.rain_mu + 2.0)
+        )
+        precomputed_evaporation_beta_exp_coeff: ta.wpfloat = (2.0 * self.config.rain_mu + 5.5) / (
+            2.0 * self.config.rain_mu + 8.0
+        ) - precomputed_evaporation_alpha_exp_coeff
+        precomputed_evaporation_beta_coeff: ta.wpfloat = (
+            0.26
+            * math.sqrt(
+                microphy_const.ref_air_density * 130.0 / microphy_const.air_kinemetic_viscosity
+            )
+            * _ar ** (-precomputed_evaporation_beta_exp_coeff)
+            * math.gamma((2.0 * self.config.rain_mu + 5.5) / 2.0)
+            / math.gamma(self.config.rain_mu + 2.0)
+        )
+
+        # Precomputations for optimization
+        power_law_exponent_for_rain_mean_fall_speed_ln1o2: ta.wpfloat = math.exp(
+            power_law_exponent_for_rain_mean_fall_speed * math.log(0.5)
+        )
+        power_law_exponent_for_ice_mean_fall_speed_ln1o2: ta.wpfloat = math.exp(
+            microphy_const.power_law_exponent_for_ice_mean_fall_speed * math.log(0.5)
+        )
+        power_law_exponent_for_graupel_mean_fall_speed_ln1o2: ta.wpfloat = math.exp(
+            microphy_const.power_law_exponent_for_graupel_mean_fall_speed * math.log(0.5)
+        )
+
+        self._ice_collision_precomputed_coef = (
+            precomputed_riming_coef,
+            precomputed_agg_coef,
+            precomputed_snow_sed_coef,
+        )
+        self._rain_precomputed_coef = (
+            power_law_exponent_for_rain_mean_fall_speed,
+            power_law_coeff_for_rain_mean_fall_speed,
+            precomputed_evaporation_alpha_exp_coeff,
+            precomputed_evaporation_alpha_coeff,
+            precomputed_evaporation_beta_exp_coeff,
+            precomputed_evaporation_beta_coeff,
+        )
+        self._sed_dens_factor_coef = (
+            power_law_exponent_for_rain_mean_fall_speed_ln1o2,
+            power_law_exponent_for_ice_mean_fall_speed_ln1o2,
+            power_law_exponent_for_graupel_mean_fall_speed_ln1o2,
+        )
+
+    @property
```

Comment:
Do you need access to these values from the outside? If not I would remove the property, internally you can simply use the private `self._ice_collision_precomputed_coeff`

Reply:
That applies to all the `@property` that you define here.